pkg/api/v1/conversion.go

```diff
@@ -510,17 +512,13 @@ func Convert_api_Pod_To_v1_Pod(in *api.Pod, out *Pod, s conversion.Scope) error
 			return err
 		}
 		out.Annotations[PodInitContainersAnnotationKey] = string(value)
-	} else {
-		delete(out.Annotations, PodInitContainersAnnotationKey)
```

Comment:
@smarterclayton - this is not needed, because delete is already called 10 lines above.

However, I'm not sure why it is causing panics.

Basically, if both InitContainers and InitContainerStatuses  are empty, the we don't copy annotations, so in this case we will try to delete annotations from the original object.
Which is obviously bad - but can it cause those panics?